File: 2020_07_20_Algorithm/src/com/company/coin0_11047.java
package com.company;

import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class coin0_11047 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int cnt = 0, won = 0;

        cnt = sc.nextInt();
        won = sc.nextInt();

        Integer[] coin = new Integer[cnt];


        for (int i = 0; i < coin.length; i++){
            coin[i] = sc.nextInt();
        }
        int result = min_count(won, coin);
        System.out.println(result);


    }
    public static int min_count(int value, Integer[] coin_list){
        int total_coin_count = 0;
        Arrays.sort(coin_list, Collections.reverseOrder());

        for (int i = 0; i < coin_list.length; i++){
            int coin_num = value/coin_list[i];
            total_coin_count += coin_num;
            value -= coin_num*coin_list[i];
        }
        return total_coin_count;
    }
}

File: 2020_07_20_Algorithm/coin0_11047.py
'''백준 11047 동전 0'''
cnt, won = map(int, input().split(" "))
coin = []
# 입력한 개수만큼의 동전종류를 입력
for i in range(cnt):
    coin.append(int(input()))
# 최소한의 동전값을 찾아주는 함수
def min_count(value, coin_list):
    total_coin_count = 0
    # 가장 큰 값부터 정렬하여
    coin_list.sort(reverse = True)
    for coin in coin_list:
        # 가장 큰 값부터 대입하여 몫을 구하고
        coin_num = value // coin
        # 구한 값으로 카운트
        total_coin_count += coin_num
        # 코인값 만큼 총 값에서 빼주고 다시 반복합니다.
        value -= coin_num * coin
    return total_coin_count

result = min_count(won, coin)
print(result)